misc/e2e_multinode/genesis.go

```diff
@@ -0,0 +1,212 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gnolang/gno/gno.land/pkg/gnoland"
+	"github.com/gnolang/gno/gnovm/pkg/gnoenv"
+	abci "github.com/gnolang/gno/tm2/pkg/bft/abci/types"
+	signer "github.com/gnolang/gno/tm2/pkg/bft/privval/signer/local"
+	bft "github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/stretchr/testify/require"
+)
+
+// createSharedGenesis creates a genesis file shared by all nodes
+func createSharedGenesis(t TestingT, tempDir string, validators []*Node) {
+	// Read all validators' private keys
+	validatorKeys := make([]*signer.FileKey, len(validators))
+	for i, validator := range validators {
+		validatorKeyPath := filepath.Join(validator.DataDir, "secrets", defaultValidatorKeyName)
+		validatorFileKey, err := signer.LoadFileKey(validatorKeyPath)
+		require.NoError(t, err, "failed to load validator key")
+		validatorKeys[i] = validatorFileKey
+	}
+
+	// Create genesis document matching official configuration
+	gen := &bft.GenesisDoc{}
+	gen.GenesisTime = time.Date(2025, 7, 25, 7, 0, 0, 0, time.UTC) // Match official genesis
+	gen.ChainID = "test7-determinism"                              // Similar to official chain ID
```

Comment:
It seems inappropriate to pretend this is related to test7 since we are not using a specific git hash for the test. This is already inaccurate today and will become even more so over time.

Reply:
using e2e_test as chainid